internal/cluster/cluster.go

```diff
@@ -223,12 +223,9 @@ func (m *Manager) RefreshStatus(ctx context.Context, c *common.Cluster, db *gorm
 	if err != nil {
 		return nil, common.NewApiError(http.StatusConflict, err)
 	}
+
 	//return updated cluster
-	var clusterAfterRefresh common.Cluster
-	if err := db.Preload("Hosts").Take(&clusterAfterRefresh, "id = ?", c.ID.String()).Error; err != nil {
```

Comment:
common.GetClusterFromDB doesn't preload hosts. But on the other hand it looks like this code doesn't really look at the hosts anyway...

Reply:
Basically, every call was replaced with the same function. `GetClusterFromDB` do preload hosts.
I didn't change any logic just replaced all the preloads calls.